### PR TITLE
Signature for selected ops

### PR DESCRIPTION
**Pull request by AlexandreEichenberger**
Up to now, signatures can be gathered at runtime only for ONNX ops, not ZHigh or others.

This PR removes the old option and replace it by `-instrument-signature=<string>` where string can be "ALL" (or ""), "NONE", or any specific op or dialect, matched by a regex, such as "onnx.*"

### CODE REVIEW

**Comment by AlexandreEichenberger** [comment]
Example of output

```
OMP_NUM_THREADS=1 ONNX_MLIR_INSTRUMENT_FILE=runNNPA6-seq-zdnn-sig2.log numactl -C 4-10 RunONNXModel.py -m roberta-base-11.onnx -c "-O3 -mcpu=z16 -maccel=NNPA -opt-report=Simd  --instrument-signature=ALL" --shape-info=0:6x384  --log-to-file compileNNPA6-seq-zdnn-sig2.log -w 2 -n 10
[...]

make-report.py -s sig -c runNNPA6-seq-zdnn-sig2.log  -l 2
[...]
  onnx.Transpose, 576
    144: sig, 6x12x384x64xfloat, 6x384x12x64xfloat
    288: sig, 6x384x12x64xfloat, 6x12x384x64xfloat
    144: sig, 6x384x12x64xfloat, 6x12x64x384xfloat
  zhigh.MatMul, 1164
    12: sig, 1x12x1x1x32x64x_Float16, 1x12x1x24x32x64x_Float16, 1x12x1x1x32x64x_Float16, 1x12x1x1x32x64x_Float16
    576: sig, 6x12x1x12x32x64x_Float16, 1x12x1x24x32x64x_Float16, 1x12x1x1x32x64x_Float16, 6x12x1x12x32x64x_Float16
    144: sig, 6x12x1x12x32x64x_Float16, 1x48x1x24x32x64x_Float16, 1x48x1x1x32x64x_Float16, 6x48x1x12x32x64x_Float16
    144: sig, 6x48x1x12x32x64x_Float16, 1x12x1x96x32x64x_Float16, 1x12x1x1x32x64x_Float16, 6x12x1x12x32x64x_Float16
    144: sig, 72x1x1x12x32x64x_Float16, 72x6x1x2x32x64x_Float16, 72x6x1x12x32x64x_Float16
    144: sig, 72x6x1x12x32x64x_Float16, 72x1x1x12x32x64x_Float16, 72x1x1x12x32x64x_Float16
  zhigh.Softmax, 144
    *: sig, 72x6x1x12x32x64x_Float16, 72x6x1x12x32x64x_Float16
  zhigh.Stick, 1020
    144: sig, 6x384x3072xfloat, 6x48x1x12x32x64x_Float16
    432: sig, 6x384x768xfloat, 6x12x1x12x32x64x_Float16
    12: sig, 6x768xfloat, 1x12x1x1x32x64x_Float16
    288: sig, 72x384x64xfloat, 72x1x1x12x32x64x_Float16
    144: sig, 72x64x384xfloat, 72x6x1x2x32x64x_Float16
  zhigh.StickifiedConstant, 1752
    732: sig, 1x12x1x1x32x64x_Float16
    588: sig, 1x12x1x24x32x64x_Float16
    144: sig, 1x12x1x96x32x64x_Float16
    144: sig, 1x48x1x1x32x64x_Float16
    144: sig, 1x48x1x24x32x64x_Float16
  zhigh.Tanh, 12
    *: sig, 1x12x1x1x32x64x_Float16, 1x12x1x1x32x64x_Float16
  zhigh.Unstick, 1308
    12: sig, 1x12x1x1x32x64x_Float16, 6x768xfloat
    720: sig, 6x12x1x12x32x64x_Float16, 6x384x768xfloat
    144: sig, 6x48x1x12x32x64x_Float16, 6x384x3072xfloat
    144: sig, 72x1x1x12x32x64x_Float16, 72x384x64xfloat
    288: sig, 72x6x1x12x32x64x_Float16, 72x384x384xfloat
```

File: src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp [hunk 1]
```diff
@@ -607,6 +607,7 @@ struct OnnxToKrnlLoweringConfiguration {
   static int reportOnSimd;
   static std::string defaultSimdComment;
   static EnableByRegexOption enableSpecificParallelOps;
+  static EnableByRegexOption traceSpecificOpSignatures;
```
Comment (tungld) [inline src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp:610]: It looks this variable should be inside InstrumentONNXSignaturePass. Also, this pass does not use it.

Comment (AlexandreEichenberger) [reply src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp:610]: I tried to move it inside but ran into issue. Let me try again, as I agree it would be the better solution.

Comment (AlexandreEichenberger) [reply src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp:610]: I found a better way to implement your recommendation than what I did before. Now it's included, no global state anymore. thanks for the suggestion, it was the right call.

Comment (tungld) [reply src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp]: Thank you!

**Review by tungld — APPROVED** [review]
LGTM.
